app/models/live_metric.rb

```diff
@@ -0,0 +1,112 @@
+class LiveMetric < ActsAsArModel
+  set_columns_hash(:timestamp => :datetime)
+
+  class LiveMetricError < RuntimeError; end
+
+  def self.where(*)
+    raise LiveMetricError, "where method is not supported on LiveMetric"
+  end
+
+  def self.find(*args)
+    validate_args(args)
+    raw_query = args[1]
+    processed = process_conditions(raw_query[:conditions])
+    resource = fetch_resource(processed[:resource_type], processed[:resource_id])
+    if resource.nil?
+      []
+    else
+      filter_and_fetch_metrics(resource, raw_query[:ext_options][:only_cols], processed[:start_time],
+                               processed[:end_time], processed[:interval_name])
+    end
+  end
+
+  def self.validate_args(args)
+    unless args && args.length > 1
+      raise LiveMetricError, "LiveMetrics expression wrong number of arguments"
```

Comment:
yes, I understand regarding Java patterns in mind, it takes time to adjust to other programming languages, especially with such different concept (dynamic vs static) and it's a natural process.
What I meant here is that typically if a method in Ruby receives 
X parameters, and you pass more, you'll get the aforementioned error. Same would happen if you would pass less parameters than expected (unless some of the parameters are non mandatory or they have a default value as part of the method signature)